code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm

```diff
@@ -0,0 +1,1195 @@
+				yum *= 2
+				handle_affinity(user, 5)
+			else if(!liked && disliked)
+				user.visible_message("<span class='notice'>\The [user] feeds \the [O] to \the [src]. It nibbles \the [O] slowly.</span>","<span class='notice'>You feed \the [O] to \the [src]. It nibbles \the [O] slowly.</span>")
+				yum *= 0.5
+				handle_affinity(user, -5)
+			else
+				user.visible_message("<span class='notice'>\The [user] feeds \the [O] to \the [src]. It nibbles \the [O].</span>","<span class='notice'>You feed \the [O] to \the [src]. It nibbles \the [O].</span>")
+				handle_affinity(user, 1)
+		else
+			user.visible_message("<span class='notice'>\The [user] feeds \the [O] to \the [src]. It nibbles \the [O] casually.</span>","<span class='notice'>You feed \the [O] to \the [src]. It nibbles \the [O] casually.</span>")
+		adjust_nutrition(yum) //add the nutriment!
+		user.drop_from_inventory(O)
+		qdel(O)
+		playsound(src, 'sound/items/eatfood.ogg', 75, 1)
+		if(!client && lets_eat(user) && prob(1))
+			visible_message("<span class='danger'>\The [src] scromfs \the [user] along with the food!</span>!")
+			to_chat(user, "<span class='notice'>\The [src] leans in close, spreading its jaws in front of you. A hot, humid gust of breath blows over you as the weight of \the [src]'s presses you over, knocking you off of your feet as the warm gooey tough of jaws scromf over your figure, rapidly guzzling you away with the [O], leaving you to tumble down into the depths of its body...</span>")
+			playsound(src, pick(bodyfall_sound), 75, 1)
+			teppi_pounce(user)
+		if(yum && nutrition >= 500)
+			to_chat(user, "<span class='notice'>\The [src] seems satisfied.</span>")
+		return
+	/////TOOLS AND WEAPONS/////
+	if(istype(O, /obj/item/weapon/tool/wirecutters))
+		if(teppi_wool)
+			if(do_after(user, 3 SECONDS, exclusive = TASK_USER_EXCLUSIVE, target = src))
+				amount_grown = rand(0,250)
+				var/obj/item/stack/material/fur/F = new(get_turf(user))
+				F.amount = rand(10,15)
+				F.color = marking_color
+				teppi_wool = FALSE
+				update_icon()
+				handle_affinity(user, 5)
+				return		
+	if(istype(O, /obj/item/weapon/material/knife))
+		var/obj/item/weapon/material/knife/K = O
+		if(teppi_wool)
+			var/sheartime = 3 SECONDS
+			if(K.default_material == MAT_PLASTIC || K.default_material == MAT_FLINT)
+				sheartime *= 2
+			if(K.dulled == TRUE)
+				sheartime *= 3
+			user.visible_message("<span class='notice'>\The [user] shears \the [src] with \the [O].</span>","<span class='notice'>You shear \the [src] with \the [O].</span>")
+			if(do_after(user, sheartime, exclusive = TASK_USER_EXCLUSIVE, target = src))
+				amount_grown = rand(0,250)
+				var/obj/item/stack/material/fur/F = new(get_turf(user))
+				F.amount = rand(10,15)
+				F.color = marking_color
+				teppi_wool = FALSE
+				update_icon()
+				handle_affinity(user, 5)
+				return
```

Comment:
Suggestion: Duplicated code could be moved to a `/proc/shear(user, tool)` proc or somesuch.